Issue 8657: Wrong name for calinski_harabaz_score

waynehamadi opened the pull request:
#### Reference Issue
#8657 

#### What does this implement/fix? Explain your changes.
It should solve the issue 8657.
I have replaced every file that contains "harabaz" by "harabasz".

#### Any other comments?
Did I forget a file?

waynehamadi committed:
Issue 8657

It should solve the issue 8657.
I have replaced every file that contains "harabaz" by "harabasz".

Did I forget a file?

waynehamadi committed:
Merge pull request #1 from MerwaneHAMADI/8657

Issue 8657

MerwaneHAMADI commented:
Travis CI failed:
./sklearn/metrics/__init__.py:45:1: F401 '.cluster.calinski_harabasz_score' imported but unused
from .cluster import calinski_harabasz_score

So I looked at ./sklearn/metrics/cluster/__init__.py and indeed:
![image](https://cloud.githubusercontent.com/assets/9652976/24429392/1ff211b4-1412-11e7-820f-8037cf51d0ba.png)

calinski_harabasz_score is imported but unused.
What I find weird is that on spyder when I type this:
![image](https://cloud.githubusercontent.com/assets/9652976/24429426/4b329952-1412-11e7-9f00-216d9ff98504.png)

the compiler doesn't say: calinski_harabasz_score is imported but unused.
I have a possible explanation:
Travis CI doesn't care about calinski_harabasz_score put into string. It considers this string like any other string and sends a failure message because an imported object unused is suspect.
What I finf weird is that if you delete the line of importation, the compiler is not ok.
![image](https://cloud.githubusercontent.com/assets/9652976/24429592/094936da-1413-11e7-91c8-65eebe961e86.png)


So, does someone know how to make work Travis CI?

jnothman commented on doc/modules/classes.rst:
```diff
@@ -881,7 +881,7 @@ details.
    metrics.adjusted_mutual_info_score
    metrics.adjusted_rand_score
-   metrics.calinski_harabaz_score
+   metrics.calinski_harabasz_score
```
at the bottom of this page there is a list of deprecated functions etc. Please add the misspelled version there.

jnothman commented on sklearn/metrics/cluster/__init__.py:
```diff
@@ -19,12 +19,12 @@
            "homogeneity_score", "mutual_info_score", "v_measure_score",
            "fowlkes_mallows_score", "entropy", "silhouette_samples",
-           "silhouette_score", "calinski_harabaz_score", "consensus_score"]
+           "silhouette_score", "calinski_harabasz_score", "consensus_score"]
```
During deprecation we should keep both the misspelled and corrected versions here

jnothman commented on sklearn/metrics/cluster/unsupervised.py:
```diff
@@ -207,13 +207,13 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
     return sil_samples
 
 
-def calinski_harabaz_score(X, labels):
```
please keep something along the lines of

```python
@deprecated(...)
def calinski_harabaz_score(X, labels):
    return calinski_harabasz_score(X, labels)
```

See http://scikit-learn.org/stable/developers/contributing.html#deprecation

jnothman reviewed (commented):
Thanks for working on this.

jnothman commented:
The imported but unused message, I suspect, may result from us only running flake8 on the diff produced by your changes.

waynehamadi committed:
Update classes.rst

waynehamadi committed:
Update __init__.py

waynehamadi committed:
Update unsupervised.py

I am not sure if that the intended behavior. If someone types:

result=calinski_harabaz_score(X, labels)  
print(result)

The "deprecated" message will appear but no results will be shown.
Is that what we want?

waynehamadi committed:
Update classes.rst

I have  replaced:
sklearn.metrics.calinski_harabasz_score
metrics.calinski_harabasz_score
to see if the commit fails or not.

waynehamadi committed:
Update __init__.py

I had forgotten to import calinski_harabaz_score

waynehamadi committed:
Update classes.rst

my mistake:
I have put metrics.calinski_harabasz_score.
I have now replaced it by metrics.calinski_harabaz_score

waynehamadi committed:
Update unsupervised.py

circle-CI error message:
name 'deprecated' is not defined.

I forgot to import deprecated. Now it should be ok.

waynehamadi committed:
Update unsupervised.py

Travis CI sent:
E304 blank lines found after function decorator

Now I have deleted the line

waynehamadi committed:
Update unsupervised.py

Travis CI sent:
./sklearn/metrics/cluster/unsupervised.py:265:1: E304 blank lines found after function decorator
I forgot to delete a line. Now it should be ok.

waynehamadi committed:
Update unsupervised.py

E501 line too long (89 > 79 characters).

Now it should be ok

LisaThomas9 commented:
I will work on this
